platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -292,14 +304,43 @@ private void captureTimestamp(final ConsensusEvent event) {
      *
      * @param transaction the transaction to apply
      */
-    private void handleTransaction(final ConsensusTransaction transaction) {
-        if (transaction.isSystem()) {
-            return;
+    private ConsensusTransaction handleTransaction(final ConsensusTransaction transaction) {
+        if (isSystemTransaction(transaction)) {
+            return transaction;
         }
+
         final int delta =
                 ByteUtils.byteArrayToInt(transaction.getApplicationTransaction().toByteArray(), 0);
         runningSum += delta;
         setChild(RUNNING_SUM_INDEX, new StringLeaf(Long.toString(runningSum)));
+
+        return null;
+    }
+
+    private boolean isSystemTransaction(final ConsensusTransaction transaction) {
+        if (transaction.isSystem()) {
+            return true;
+        }
+
+        try {
+            final var transactionBytes = transaction.getApplicationTransaction();
+
+            if (Bytes.EMPTY.equals(transaction.getApplicationTransaction())) {
+                return false;
+            }
+
+            final var parsedTransaction =
+                    StateSignatureTransaction.PROTOBUF.parseStrict(transactionBytes.toReadableSequentialData());
+
+            if (parsedTransaction.signature() != Bytes.EMPTY) {
+                return true;
+            }
+        } catch (ParseException e) {
+            logger.error("Failed to parse transaction body", e);
+            return false;
+        }
```

Comment:
throwing and catching an exception is quite costly for resources, and this method tries to parse every app transaction as if its a system transaction. we should instead have a way to tell if its app or system before trying to parse it

Reply:
Agreed - switched the logic to only check bytes length. StateSignatureTransaction will definitely have length bigger than 4 (the random application bytes used in this test)